src/main/java/org/sqlite/core/CoreStatement.java

```diff
@@ -119,5 +131,11 @@ protected void internalClose() throws SQLException {
         if (resp != SQLITE_OK && resp != SQLITE_MISUSE) conn.getDatabase().throwex(resp);
     }
 
+    protected void notifyFirstStatementExecuted() {
+        if(this.conn instanceof JDBC3Connection){
```

Comment:
unnecessary check as `setFirstStatementWasExecuted` is defined on `SQLiteConnection`